t/t3431-rebase-ignore-ws.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+#
+# Copyright (c) 2019 Rohit Ashiwal
+#
+
+test_description='git rebase --ignore-whitespace tests'
+
+. ./test-lib.sh
+
+cat >patch1.patch <<EOF
+diff --git a/main.c b/main.c
+new file mode 100644
+--- /dev/null
++++ b/main.c
+@@ -0,0 +1,6 @@
++#include <stdio.h>
++
++int main() {
++	printf("Hello World\n");
++	return 0;
++}
+EOF
+
+sed -e 's/Z/ /g' >patch2.patch <<EOF
+diff --git a/main.c b/main.c
+--- a/main.c
++++ b/main.c
+@@ -3,4 +3,4 @@
+Zint main() {
+-	printf("Hello World\n");
++	printf("Hello Git\n");
+Z	return 0;
+Z}
+EOF
+
+sed -e 's/	/        /g' <patch1.patch >patch3.patch
+
+# We assume that git-apply is tested independently and works as usual :)
+
+test_expect_success 'setup' '
+	git commit --allow-empty -m "initial" &&
+	git checkout -b topic &&
+	git apply patch1.patch &&
```

Comment:
Rather than using `git apply` here, I think it would be simpler to just create the file and use `git add` + `git commit` to add it.   And then for the subsequent changes a simple `sed` would suffice.